warehouse/query-core/src/main/java/datawave/query/function/DocumentScanRangeProvider.java

```diff
@@ -0,0 +1,47 @@
+package datawave.query.function;
+
+import org.apache.accumulo.core.data.Key;
+import org.apache.accumulo.core.data.PartialKey;
+import org.apache.accumulo.core.data.Range;
+
+/**
+ * Given arbitrary "Document Key", provides the scan range that covers the document.
+ */
+public class DocumentScanRangeProvider implements ScanRangeProvider {
```

Comment:
Is Document how we want to refer to this? Event data is the only thing this works on. Scan is probably redunant... Consider EventRangeProvider :)

Reply:
Agree that scan is probably redundant. Here are some thoughts on Document vs. Event, in favor of keeping the Document naming convention
- KeyToDocumentData is not called KeyToEventData
- we are operating on keys explicitly named "document keys"
- TLD is top level document, not top level event
- TLD extends this class, it would seem weird to see `TopLevelDocumentRangeProvider extends EventRangeProvider`...but maybe not?

Few thoughts in favor of renaming things to be Event
- we have EventDataFilters, not "DocumentDataFilters"
- we have TLDEventDataFilters (argument in favor of TLD extending something named Event)

At best we can conclude naming conventions are inconsistent. I'm leaning towards removing "scan" from the names, but leaving "document" in place given the context of 'documentKey' and 'KeyToDocumentData'. Let me know your thoughts.